File: flutter/lib/data/models/bus.dart
class Bus {
  final int IdBus;
  final String placa;
  final int? capacidad;
  final String? modelo;
  final String? marca;
  final String? nombre;
  final String? numero;
  final int? ChoferId;
  final int? RutaId;

  const Bus({
    required this.IdBus,
    required this.placa,
    this.capacidad,
    this.modelo,
    this.marca,
    this.nombre,
    this.numero,
    this.ChoferId,
    this.RutaId,
  });

  factory Bus.fromJson(Map<String, dynamic> json) {
    return Bus(
      IdBus: json['IdBus'] ?? 0,
      placa: json['placa'] ?? "", // Cambiado de 'Placa'
      capacidad: json['capacidad'], // Cambiado de 'Capacidad'
      modelo: json['modelo'], // Cambiado de 'Modelo'
      marca: json['marca'], // Cambiado de 'Marca'
      nombre: json['nombre'], // Nuevo campo
      numero: json['numero'], // Nuevo campo
      ChoferId: json['ChoferId'],
      RutaId: json['RutaId'],
    );
  }

  Map<String, dynamic> toJson() => {
    'IdBus': IdBus,
    'placa': placa, // Cambiado
    'capacidad': capacidad, // Cambiado
    'modelo': modelo, // Cambiado
    'marca': marca, // Cambiado
    'nombre': nombre, // Nuevo campo
    'numero': numero, // Nuevo campo
    'ChoferId': ChoferId,
    'RutaId': RutaId,
  };
}

File: Backend/models/bus.py
# from sqlmodel import SQLModel, Field, Relationship
# from typing import Optional
# import datetime
# from sqlalchemy import Column, DateTime

# class BusBase(SQLModel):
#     placa: str
#     capacidad: Optional[int] = 40
#     modelo: Optional[str] = None
#     marca: Optional[str] = None

# class Bus(BusBase, table=True):
#     IdBus: Optional[int] = Field(default=None, primary_key=True)
#     ChoferId: Optional[int] = Field(default=None, foreign_key="chofer.IdChofer")
#     RutaId: Optional[int] = Field(default=None, foreign_key="ruta.IdRuta")

#     chofer: Optional["Chofer"] = Relationship(back_populates="buses")
#     ruta: Optional["Ruta"] = Relationship(back_populates="buses")
#     favoritos: Optional["BusFavorito"] = Relationship(back_populates="bus")
# paradero: Optional["Paradero"] = Relationship(back_populates="buses")

# class BusCreate(BusBase):
#     ChoferId: Optional[int] = None
#     RutaId: Optional[int] = None

# class BusPublic(BusBase):
#     IdBus: int
#     ChoferId: Optional[int]
#     RutaId: Optional[int]

# class BusUpdate(SQLModel):
#     placa: Optional[str] = None
#     capacidad: Optional[int] = None
#     RutaId: Optional[int] = None
#     ChoferId: Optional[int] = None
#     modelo: Optional[str] = None
#     marca: Optional[str] = None




# models/bus.py - VERSIÓN CORREGIDA
from sqlmodel import SQLModel, Field, Relationship
from typing import Optional
import datetime
from sqlalchemy import Column, DateTime

class BusBase(SQLModel):
    placa: str
    capacidad: Optional[int] = 40
    modelo: Optional[str] = None
    marca: Optional[str] = None
    nombre:Optional[str]=None
    numero:Optional[str]=None

class Bus(BusBase, table=True):
    IdBus: Optional[int] = Field(default=None, primary_key=True)
    ChoferId: Optional[int] = Field(default=None, foreign_key="chofer.IdChofer")
    RutaId: Optional[int] = Field(default=None, foreign_key="ruta.IdRuta")

    chofer: Optional["Chofer"] = Relationship(back_populates="buses")
    ruta: Optional["Ruta"] = Relationship(back_populates="buses")
    paradero: Optional["Paradero"] = Relationship(back_populates="bus")
    busfavorito:Optional["BusFavorito"]= Relationship(back_populates="bus")

class BusCreate(BusBase):
    ChoferId: Optional[int] = None
    RutaId: Optional[int] = None

class BusPublic(BusBase):
    IdBus: int
    ChoferId: Optional[int]
    RutaId: Optional[int]

class BusUpdate(SQLModel):
    placa: Optional[str] = None
    capacidad: Optional[int] = None
    RutaId: Optional[int] = None
    ChoferId: Optional[int] = None
    modelo: Optional[str] = None
    marca: Optional[str] = None